node/bft/src/gateway.rs

```diff
@@ -561,6 +566,21 @@ impl<N: Network> Gateway<N> {
                     return Ok(());
                 }
             }
+            Event::PrimaryPing(_) | Event::BatchCertified(_) => {
+                // Retrieve the certificate ID.
+                let certificate_id = match &event {
+                    Event::PrimaryPing(PrimaryPing { certificate_id, .. }) => *certificate_id,
+                    Event::BatchCertified(BatchCertified { certificate_id, .. }) => *certificate_id,
+                    _ => unreachable!(),
+                };
+                // Skip processing duplicate primary certificates received from the same peer.
+                let num_events = self
+                    .cache
+                    .insert_inbound_primary_certificate((peer_ip, certificate_id), CACHE_PRIMARY_CERTIFICATES_INTERVAL);
+                if num_events >= 2 {
```

Comment:
Current attributes are (for messages with the same certiifcate id):

- Only 1 can be processed at a time
- Can be processed again after 15 seconds

How can we dial this number to be safe? I feel like malicious validators might be able to prevent other nodes from syncing certificates by:

1. Send a certain batch certificate to an honest node.
2. Withholding the internal data to prevent the honest node from accurately processing it.
3. Repeat every 15 seconds OR send many of the same message to fill up the cache.
4. This will prevent the honest node from processing the certificate from other honest validators.